### ISSUE TITLE
clang: 'argument unused during compilation: '-rtlib=compiler-rt' warning when using -nostdlib

### ISSUE THREAD

**Issue by nolange** [llvm/llvm-project#64939]
Hello,

I have a cmake project which includes some kinda plugins that need to be built with `-nostdlib`.
Using a toochain-file for clang which adds `-rtlib=compiler-rt` for linker steps will result in wrong warnings,
for the plugins as those 2 flags are then used together.

Small reproducer:

``` bash
echo "void foo() {}" > test.c
clang -c -o test.o test.c
clang -rtlib=compiler-rt -nostdlib --shared test.o
```

```
clang: warning: argument unused during compilation: '-rtlib=compiler-rt' [-Wunused-command-line-argument]
```

I believe this to be a bug, as

-   I want to use `-rtlib=compiler-rt` if an runtime is needed (defined in a toolchain file for the whole build),
    making this flag optional in regard to other options would be a compiler-specific mess.
-   setting `-nostdlib++` and `-stdlib=libc++` is similar and doesnt result in a warning.

**Comment by MaskRay** [comment]
https://maskray.me/blog/2023-08-25-clang-wunused-command-line-argument There is a tension between `-Wunused-command-line-argument` and default options.

> setting -nostdlib++ and -stdlib=libc++ is similar and doesnt result in a warning.

There are many other options that do result in diagnostics:
```
% clang --rtlib=compiler-rt --unwindlib=libunwind -stdlib=libstdc++ -static-libstdc++ -pthread -nostdlib a.o
clang: warning: argument unused during compilation: '--rtlib=compiler-rt' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '--unwindlib=libunwind' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-static-libstdc++' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-pthread' [-Wunused-command-line-argument]
```

I mentioned the solutions 

> To suppress the diagnostics, we can utilitize [--start-no-unused-arguments and --end-no-unused-arguments](https://reviews.llvm.org/D116503) (Clang 14) like the following:
